pkg/log/log.go

```diff
@@ -1,84 +1,372 @@
+func (e *Event) Interface(key string, val interface{}) *Event {
+	e.attrs = append(e.attrs, slog.Any(key, val))
+
+	return e
+}
+
+// Any adds any interface field to the event (alias for Interface).
+func (e *Event) Any(key string, val interface{}) *Event {
+	return e.Interface(key, val)
+}
+
+// Strs adds a slice of strings field to the event.
+func (e *Event) Strs(key string, vals []string) *Event {
+	e.attrs = append(e.attrs, slog.Any(key, vals))
+
+	return e
+}
+
+// IPAddr adds an IP address field to the event.
+func (e *Event) IPAddr(key string, ip interface{}) *Event {
+	e.attrs = append(e.attrs, slog.String(key, fmt.Sprintf("%v", ip)))
+
+	return e
+}
+
+// RawJSON adds a raw JSON field to the event.
+func (e *Event) RawJSON(key string, data []byte) *Event {
+	e.attrs = append(e.attrs, slog.String(key, string(data)))
+
+	return e
+}
+
+// Dur adds a duration field to the event.
+func (e *Event) Dur(key string, d time.Duration) *Event {
+	e.attrs = append(e.attrs, slog.Duration(key, d))
+
+	return e
+}
+
+// NewTestLogger creates a logger for testing purposes (replaces zerolog.New(os.Stdout)).
+func NewTestLogger() Logger {
+	return NewLogger("debug", "")
+}
+
+// NewTestLoggerPtr creates a pointer to a logger for testing purposes.
+func NewTestLoggerPtr() *Logger {
+	logger := NewLogger("debug", "")
+
+	return &logger
+}
+
+// Msgf logs the event with a formatted message.
+func (e *Event) Msgf(format string, args ...interface{}) {
+	msg := fmt.Sprintf(format, args...)
+	e.logger.LogAttrs(nil, e.level, msg, e.attrs...)
+
+	if e.isPanic {
+		panic(msg)
 	}
+}
 
-	zerolog.SetGlobalLevel(lvl)
+// Msg logs the event with a simple message.
+func (e *Event) Msg(msg string) {
+	e.logger.LogAttrs(nil, e.level, msg, e.attrs...)
 
-	var log zerolog.Logger
+	if e.isPanic {
+		panic(msg)
+	}
+}
+
+func (l Logger) Println(v ...interface{}) {
+	l.Error().Msg("panic recovered") //nolint: check-logs
+	log.Println(v...)
+}
 
+func (l Logger) Printf(format string, a ...interface{}) {
+	log.Printf(format, a...)
+}
+
+// parseLevel converts string level to slog.Level.
+func parseLevel(level string) (slog.Level, error) {
+	switch strings.ToLower(level) {
+	case "debug":
+		return slog.LevelDebug, nil
+	case "info":
+		return slog.LevelInfo, nil
+	case "warn", "warning":
+		return slog.LevelWarn, nil
+	case "error":
+		return slog.LevelError, nil
+	default:
+		return slog.LevelInfo, errors.ErrBadConfig
+	}
+}
+
+func NewLogger(level, output string) Logger {
+	// Parse log level
+	lvl, err := parseLevel(level)
+	if err != nil {
+		panic(err)
+	}
+
+	// Determine output writer
+	var writer io.Writer
 	if output == "" {
-		log = zerolog.New(os.Stdout)
+		writer = os.Stdout
 	} else {
 		file, err := os.OpenFile(output, os.O_APPEND|os.O_WRONLY|os.O_CREATE, defaultPerms)
 		if err != nil {
 			panic(err)
 		}
+		writer = file
+	}
+
+	// Create JSON handler with RFC3339Nano time format
+	opts := &slog.HandlerOptions{
```

Comment:
I think we could have a common function NewJsonHandler(level, writer) which also initializes the options. It is common code in 3 functions.